platforms/software/resources-http/src/main/java/org/gradle/internal/resource/transport/http/DefaultHttpSettings.java

```diff
@@ -62,7 +64,8 @@ private DefaultHttpSettings(
         RedirectMethodHandlingStrategy redirectMethodHandlingStrategy,
         int maxRedirects,
         int maxConnTotal,
-        int maxConnPerRoute
+        int maxConnPerRoute,
+        boolean disableContentCompression
```

Comment:
❌ I don't think that compression should be disabled. It is [compatible](https://stackoverflow.com/questions/33947562/is-it-possible-to-send-http-response-using-gzip-and-byte-ranges-at-the-same-time) with ranged downloads.

This might also be the answer to the issue [you had](https://github.com/gradle/gradle/issues/30323#issuecomment-2325444002) that the file size doesn't match the requested range size.

Reply:
Reverted.